openupgradelib/openupgrade.py

```diff
@@ -643,13 +644,19 @@ def write_value(ids, field, value):
                 cr.execute(query, (params[0], sub_ids))
 
     for model in default_spec.keys():
-        obj = pool.get(model)
-        if not obj:
-            do_raise(
-                "Migration: error setting default, no such model: %s" % model)
+        if version_info[0] >= 10:
```

Comment:
No need for this `if`. It's the same for all versions: `pool/env.get(model)`

Reply:
I beg to differ with the estimated colleague @pedrobaeza. Only since 10, `Environment` inherits from `Mapping` which implements the `.get` function. In <= 9, this will raise.
As 8 doesn't even support the `in` operator (https://github.com/OCA/OCB/blob/8.0/openerp/api.py#L779), we're really out of luck here.

I think btw. it's not a good idea anymore to check for the version - we still can request a registry if we want or an env in pre 10: https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L1152 Let's better check if `pool` is an `api.Envrionment`, then you can first check `model in pool.registry` and then use the itemgetter on `pool` itself